### DISCUSSION TITLE
SLF4J 1.7 and 2.0 are not fully binary compatible

### DISCUSSION THREAD

**Discussion by jcompagner** [qos-ch/slf4j#326] [General]
In the docs i think this is said that it is a drop in replacement
and for the most part i think it is.

but there is a change:

1.7:
https://github.com/qos-ch/slf4j/blob/e9ee55cca93c2bf26f14482a9bdf961c750d2a56/slf4j-api/src/main/java/org/slf4j/event/LoggingEvent.java

2.0:
https://github.com/qos-ch/slf4j/blob/f85104040ce44ac545e15e4f41ef771a7a7f7add/slf4j-api/src/main/java/org/slf4j/event/LoggingEvent.java

so getMarker() is removed (getMarkers() is added)

shouldn't getMarker() just stay there (and maybe marked as deprecated) ?

**Comment by ennru** [comment]
Correct, this change is a bummer for upgrading.
It was introduced in [this commit](https://github.com/qos-ch/slf4j/commit/1f3b8726d160c51befdf57ec42c6bafbd8b8652c#diff-5b807b828a4ae210a6bd640c50e64b21eb15f1477fe640aec36d947ef2372c82R16).

**Comment by jcompagner** [comment]
this was already set straight in: https://jira.qos.ch/browse/SLF4J-577

so getMarker() was not a client api (so client using the logger)
but just for the implementations (log4j and so on)

**Reply by ennru** [reply]
Thank you for linking to that ticket. A pity our library made use of this functionality as the "internal use only" wasn't clear enough.
